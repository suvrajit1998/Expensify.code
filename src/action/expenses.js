import uuid from "uuid";

export const addexpense = ({
  decription = "",
  note = "",
  amount = 0,
  createdAt = 0
}) => ({
  type: "ADD_EXPENSES",
  expenses: {
    id: uuid(),
    decription,
    note,
    amount,
    createdAt
  }
});

export const removeid = ({ id } = {}) => ({
  type: "REMOVE_ID",
  id
});

export const editexpense = (id, updets) => ({
  type: "EDIT_EXPENSE",
  id,
  updets
});
