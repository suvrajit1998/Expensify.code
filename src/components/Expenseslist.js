import React from "react";
import { connect } from "react-redux";
import ExpenseslistItem from "./expenselistitem";
import Selectexpenses from "../selector/getvisibleexpenses";

export const ExpensesList = props => (
  <div>
    <h1>Expenses List</h1>
    {props.expenses.length === 0 ? (
      <p>No Expenses</p>
    ) : (
      props.expenses.map(expens => {
        return <ExpenseslistItem key={expens.id} {...expens} />;
      })
    )}
  </div>
);

const mapstatetoprops = state => {
  return {
    expenses: Selectexpenses(state.expenses, state.filters)
  };
};

export default connect(mapstatetoprops)(ExpensesList);
