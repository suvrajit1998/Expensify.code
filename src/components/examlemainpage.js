import React from "react";
import ExpensesList from "./Expenseslist";
import Expenselistfilters from "./expenseslistfilter";

export const Examlemainpage = () => (
  <div>
    This is the Main Page Content
    <Expenselistfilters />
    <ExpensesList />
  </div>
);

export default Examlemainpage;
