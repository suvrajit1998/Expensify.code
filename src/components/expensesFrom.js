import React from "react";
import moment from "moment";
import { SingleDatePicker } from "react-dates";

const now = moment();

export default class ExpensesFrom extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      decription: props.expense ? props.expense.decription : "",
      note: props.expense ? props.expense.note : "",
      amount: props.expense ? (props.expense.amount / 100).toString() : "",
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      calenderFocus: false,
      error: ""
    };
  }

  onDecription = e => {
    const decription = e.target.value;
    this.setState(() => ({ decription }));
  };

  onNotechang = e => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };

  onAmountChange = e => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };

  onDateChange = createdAt => {
    this.setState(() => ({ createdAt }));
  };

  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calenderFocus: focused }));
  };

  onSubmit = e => {
    e.preventDefault();

    if (!this.state.decription || !this.state.amount) {
      this.setState(() => ({
        error: "Please provide Description Amount"
      }));
    } else {
      this.setState(() => ({ error: "" }));
      this.props.onSubmited({
        decription: this.state.decription,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };

  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <input
            type="text"
            placeholder="Description"
            autoFocus
            value={this.state.decription}
            onChange={this.onDecription}
          />
          <input
            type="text"
            placeholder="Amount"
            value={this.state.amount}
            onChange={this.onAmountChange}
          />
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calenderFocus}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
          <textarea
            placeholder="Add a note for your Expense (Optional)"
            value={this.state.note}
            onChange={this.onNotechang}
          ></textarea>
          <button>Add Expense</button>
        </form>
      </div>
    );
  }
}
