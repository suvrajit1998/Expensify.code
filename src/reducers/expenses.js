const expensreducerdefaultarray = [];

export default (state = expensreducerdefaultarray, action) => {
  switch (action.type) {
    case "ADD_EXPENSES": {
      return [...state, action.expenses];
    }
    case "REMOVE_ID": {
      return state.filter(({ id }) => id !== action.id);
    }
    case "EDIT_EXPENSE": {
      return state.map(expenses => {
        if (expenses.id === action.id) {
          return {
            ...expenses,
            ...action.updets
          };
        } else {
          return expenses;
        }
      });
    }
    default:
      return state;
  }
};
