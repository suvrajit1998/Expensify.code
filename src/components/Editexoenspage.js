import React from "react";
import { connect } from "react-redux";
import Expenseform from "./expensesFrom";
import { editexpense, removeid } from "../action/expenses";

const Editexpenspage = props => {
  console.log(props);
  return (
    <div>
      <Expenseform
        expense={props.expense}
        onSubmited={expense => {
          props.dispatch(editexpense(props.expense.id, expense));
          props.history.push("/");
        }}
      />
      <button
        onClick={() => {
          props.dispatch(removeid({ id: props.expense.id }));
          props.history.push("/");
        }}
      >
        Remove
      </button>
    </div>
  );
};

const tomatchprops = (state, props) => {
  return {
    expense: state.expenses.find(
      expense => expense.id === props.match.params.id
    )
  };
};

export default connect(tomatchprops)(Editexpenspage);
