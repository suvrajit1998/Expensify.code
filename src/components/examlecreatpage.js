import React from "react";
import Expensesfrom from "./expensesFrom";
import { addexpense } from "../action/expenses";
import { connect } from "react-redux";

export class Examlecreatpage extends React.Component {
  onSubmited = expense => {
    this.props.addexpense(expense);
    this.props.history.push("/");
  };

  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <Expensesfrom onSubmited={this.onSubmited} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    addexpense: expense => dispatch(addexpense(expense))
  };
};

export default connect(undefined, mapDispatchToProps)(Examlecreatpage);
