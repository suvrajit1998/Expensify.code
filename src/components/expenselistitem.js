import React from "react";
import { connect } from "react-redux";
import { removeid } from "../action/expenses";

const ExpensesItem = ({ decription, amount, createdAt, id, dispatch }) => (
  <div>
    <h3>{decription}</h3>
    <p>
      {amount} - {createdAt}
    </p>
    <button
      onClick={() => {
        dispatch(removeid({ id }));
      }}
    >
      Remove
    </button>
  </div>
);

export default connect()(ExpensesItem);
