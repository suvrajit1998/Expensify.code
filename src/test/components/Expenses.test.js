import React from "react";
import { shallow } from "enzyme";
import { ExpensesList } from "../../components/Expenseslist";
import expenses from "../fixture/expenses";

test("Should render expenses list", () => {
  const wrapper = shallow(<ExpensesList expenses={expenses} />);

  expect(wrapper).toMatchSnapshot();
});

test("Should render expense list with emty array", () => {
  const wrapper = shallow(<ExpensesList expenses={[]} />);

  expect(wrapper).toMatchSnapshot();
});
