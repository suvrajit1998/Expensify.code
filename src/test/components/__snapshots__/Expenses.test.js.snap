// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expense list with emty array 1`] = `
<div>
  <h1>
    Expenses List
  </h1>
  <p>
    No Expenses
  </p>
</div>
`;

exports[`Should render expenses list 1`] = `
<div>
  <h1>
    Expenses List
  </h1>
  <ExpensesItem
    amount={156}
    createdAt={0}
    decription="gum"
    id="1"
    key="1"
    note=""
  />
  <ExpensesItem
    amount={195}
    createdAt={-345600000}
    decription="rent"
    id="2"
    key="2"
    note=""
  />
  <ExpensesItem
    amount={29000}
    createdAt={345600000}
    decription="credit card"
    id="3"
    key="3"
    note=""
  />
</div>
`;
